### ISSUE TITLE
warning: unknown warning option '-Wno-extended-offsetof'; did you mean '-Wno-invalid-offsetof'? [-Wunknown-warning-option]

### ISSUE THREAD

**Issue by geeknik** [openssl/openssl#8859]
While compiling OpenSSL with clang version 9.0.0 (trunk 357692), I'm seeing the following messages:

```
clang -I.. -I../.. -I../modes -I../asn1 -I../evp -I../../include  -DOPENSSL_THREADS -D_REENTRANT -DDSO_DLFCN -DHAVE_DLFCN_H --debug -DPEDANTIC -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -g3 -fno-omit-frame-pointer -fsanitize=fuzzer,undefined -fno-sanitize=alignment -m64 -DL_ENDIAN -O3 -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wno-language-extension-token  -Wno-extended-offsetof -Qunused-arguments -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DRC4_ASM -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DMD5_ASM -DAES_ASM -DVPAES_ASM -DBSAES_ASM -DWHIRLPOOL_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION  -c -o set_key.o set_key.c
warning: unknown warning option '-Wno-extended-offsetof'; did you mean '-Wno-invalid-offsetof'? [-Wunknown-warning-option]
1 warning generated.
```

Replacing `-Wno-extended-offsetof` with `-Wno-invalid-offsetof` in Configure solves the problem and I don't see any errors as a result.

**Comment by mattcaswell** [comment]
Is this still an issue with current version of clang? I don't see this with my version:

````
$ clang --version
clang version 9.0.0-2 (tags/RELEASE_900/final)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
````

**Comment by mspncp** [comment]
Are you using an OpenSSL version < 1.0.2t? The `-Wno-unknown-warning-option` was added in commit

* 6a7bad0fd7a2125d075e459b33145d4ce5ee0de9 Configure: clang: add -Wno-unknown-warning-option

to suppress this warning. For an explanation, see the [commit message](https://github.com/openssl/openssl/commit/6a7bad0fd7a2125d075e459b33145d4ce5ee0de9). This commit is contained in OpenSSL 1.0.2t.

Note: on the master and OpenSSL_1_1_1-stable branch the flag was already present, but due to its importance it was moved to the front with an explaining commit message:

* ad9c296c529d3eba67a51118c8e5d875d17f6d07 Configure: clang: move -Wno-unknown-warning-option to the front [1.1.1]
* 03e5668343078b963cc6544ad7270743de13e514 Configure: clang: move -Wno-unknown-warning-option to the front [master]

**Comment by mspncp** [comment]
Sorry, I didn't notice that the question was from May. So the correct answer is probably now: should be fixed on master, 1.1.1, and 1.0.2.

**Comment by mspncp** [comment]
P.S: @geeknik I don't think you need [-Wno-invalid-offsetof](https://clang.llvm.org/docs/DiagnosticsReference.html#winvalid-offsetof) option, which happens to have a similar name but seems to be a C++ specific option.
